shaders/ssao.frag

```diff
@@ -0,0 +1,92 @@
+#version 460
+#extension GL_EXT_nonuniform_qualifier: enable
+
+#include "bindless.glsl"
+#include "scene.glsl"
+
+layout (push_constant) uniform PushConstants
+{
+    uint normalRIndex;
+    uint positionIndex;
+    uint noiseIndex;
+    uint screenWidth;
+    uint screenHeight;
+    float aoStrength;
+    float aoBias;
+    float aoRadius;
+    float minAoDistance; // Distance at which AO parameters start to adjust
+    float maxAoDistance; // Distance at which AO parameters are at full strength
+} pushConstants;
+
+layout (set = 2, binding = 0) uniform CameraUBO
+{
+    Camera camera;
+};
+
+layout (set = 1, binding = 0) buffer SampleKernel { vec4 samples[]; } uSampleKernel;
+
+layout (location = 0) in vec2 texCoords;
+layout (location = 0) out vec4 outColor;
+
+
+const int kernelSize = 64;
```

Comment:
Shouldn't this be a parameter that you pass to the shader?

Reply:
I chose to left it hard-coded 64, since the kernel is generated and cached on the CPU side with that size anyway and I don't want to regenerate it later.

Reply:
Alright, a nitpick but can you fix the formatting, it bothers me that the variable is directly above the main function XD